[JS] Storage API - path update (#7145)
aws-amplify/docs: src/pages/[platform]/build-a-backend/storage/upload/index.mdx

The `uploadData` method uploads files into Amazon S3.

<BlockSwitcher>
<Block name="With Path (Above v6.x.x)">
```javascript
import { uploadData } from 'aws-amplify/storage';

try {
  const result = await uploadData({
    path: "album/2024/1.jpg", 

>     // Alternatively, path: ({identityId}) => `album/{identityId}/1.jpg`

Reviewer: Should this go before or on same line?

Author: i thought this read better

    key: filename,
    data: file,
    options: {
      accessLevel: 'guest', // defaults to `guest` but can be 'private' | 'protected' | 'guest'

>       onProgress // Optional progress callback.

Reviewer: Should we specify an example, or can assume this symbol exists already?

```suggestion
      onProgress // Optional progress callback.
```

Author: these snippet exactly as what it is right now. we might wanna change that everywhere

    }
  }).result;
  console.log('Succeeded: ', result);
} catch (error) {
  console.log('Error : ', error);
}
```
</Block>
</BlockSwitcher>

### Monitor progress of upload

> To track the progress of your upload, you can use the `onProgress` in options:

Reviewer: ```suggestion
To track the progress of your upload, you can use the `onProgress` option:
```

<BlockSwitcher>
<Block name="With Path (Above v6.x.x)">
```javascript
import { uploadData } from 'aws-amplify/storage';

try {
  const result = uploadData({
    path: "album/2024/1.jpg", 
    // Alternatively, path: ({identityId}) => `album/{identityId}/1.jpg`
    data: file,
    options: {
      onProgress: ({ transferredBytes, totalBytes }) => {
        if (totalBytes) {
          console.log(
            `Upload progress ${
              Math.round((transferredBytes / totalBytes) * 100)

>             } %`

Reviewer: ```suggestion
            }%`
```

Author: keeping it same atm to be same as key
